mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSRepositoryAsNode.kt

```diff
@@ -62,6 +66,47 @@ data class MPSRepositoryAsNode(val repository: SRepository) : IDefaultNodeAdapte
             emptyList()
         }
     }
+
+    override fun addNewChild(role: IChildLink, index: Int, concept: IConcept?): INode {
+        throw IllegalArgumentException("Template node required for creating MPS modules")
+    }
+
+    override fun addNewChild(role: IChildLink, index: Int, concept: IConceptReference?): INode {
+        throw IllegalArgumentException("Template node required for creating MPS modules")
+    }
+
+    override fun addNewChild(role: String?, index: Int, concept: IConcept?): INode {
+        throw IllegalArgumentException("Template node required for creating MPS modules")
+    }
+
+    override fun addNewChild(role: String?, index: Int, concept: IConceptReference?): INode {
+        throw IllegalArgumentException("Template node required for creating MPS modules")
+    }
+
+    override fun addNewChild(role: IChildLink, index: Int, templateNode: INode): INode {
+        if (role.conformsTo(BuiltinLanguages.MPSRepositoryConcepts.Repository.modules)) {
+            val project = ProjectHelper.getProject(repository)
+                ?: ProjectManager.getInstance().openedProjects.firstOrNull()
+                ?: error("No MPS project available")
+            val moduleId = templateNode.getPropertyValue(BuiltinLanguages.MPSRepositoryConcepts.Module.id)
+                ?.let { ModuleId.fromString(it) } ?: ModuleId.regular()
+            val moduleName = requireNotNull(templateNode.getPropertyValue(BuiltinLanguages.jetbrains_mps_lang_core.INamedConcept.name)) { "Module has not name" }
+            val newModule = when (templateNode.concept?.getUID()) {
+                BuiltinLanguages.MPSRepositoryConcepts.Solution.getUID() -> {
+                    MPSProjectUtils.createModule(
+                        project as MPSProject,
+                        moduleName,
+                        moduleId,
+                        this,
+                    )
+                }
+                else -> throw UnsupportedOperationException("Modules of type ${templateNode.concept} not supported yet")
+            }
+            return MPSModuleAsNode(newModule)
+        } else {
+            throw UnsupportedOperationException("adding nodes to $role not implemented yet")
```

Comment:
nit: consider [returning early](https://openfolder.sh/returning-early) to make have less indentation and making code more readable.

Reply:
Refactored that method.